### PR TITLE
Fix edge cases in profile field UI migration

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

[This PR](https://github.com/forem/forem/pull/12015) which removed the `SiteConfig.dev_to?` in favor of a step towards more full extensibility essentially worked as expected, except there were some edge cases missed.

Mostly this was a result of overlooking the mapped attribute renaming in a couple places. This _should_ iron out what got missed.

This PR includes adding new SVG icons to update with the current naming convention without removing the old ones to bring redundancy in case there is still need for the old icons for any reason.

***

This PR is a patch of the current behavior. Further steps to full-on extensible generalization can include allowing profile field attributes to have an icon and creating a new display area specialized for "external links"... But that is outside the scope of this step.

## Related Tickets & Documents

https://github.com/forem/forem/pull/12015
https://github.com/forem/forem/issues/12161

## QA Instructions, Screenshots, Recordings

_Double check that this is indeed the exhaustive fix._

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: This is a bug fix on functionality which is otherwise in the communication pipeline already.

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -31,6 +31,9 @@ class Profile < ApplicationRecord
     twitch_url
     mastodon_url
     website_url
+    git_lab_url
+    linked_in_url
+    stack_overflow_url
```
Comment (Zhao-Andy) [inline app/models/profile.rb:36]: Should `dribbble_url` be here too?

File: app/models/profile.rb [hunk 2]
```diff
@@ -31,6 +31,9 @@ class Profile < ApplicationRecord
     twitch_url
     mastodon_url
     website_url
+    git_lab_url
```
Comment (jacobherrington) [inline app/models/profile.rb:34]:
```suggestion
    gitlab_url
```

This will fix the failing specs

**Review by jacobherrington — CHANGES_REQUESTED** [review]

File: lib/data_update_scripts/20210211164634_fix_profile_field_edge_cases.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module DataUpdateScripts
+  class FixProfileFieldEdgeCases
+    def run
+      ProfileField.where(attribute_name: %w[git_lab_url linked_in_url stack_overflow_url dribbble_url])
```
Comment (mstruve) [inline lib/data_update_scripts/20210211164634_fix_profile_field_edge_cases.rb:4]: should this also be `gitlab_url` here instead?

I also noticed that `stackoverflow_url` and `linkedin_url` are different in the `SPECIAL_SOCIAL_LINK_ATTRIBUTES`, is that supposed to be like that?

Comment (jacobherrington) [reply lib/data_update_scripts/20210211164634_fix_profile_field_edge_cases.rb:4]: I believe these values are actually correct. It always confuses me, but AFAIK this is how the data is actually stored. :upside_down_face: 

https://github.com/forem/forem/blob/ef481d07a33f91c2845ed8af4b51ab5cef217315/app/models/profile.rb#L38

**Review by jacobherrington — APPROVED** [review]
Works locally!

File: lib/data_update_scripts/20210211164634_fix_profile_field_edge_cases.rb (hunk 3 above)
Comment (benhalpern) [reply lib/data_update_scripts/20210211164634_fix_profile_field_edge_cases.rb:4]: Yeah, these are correct, and that is actually what is prompting this PR. The whole thing is pretty unintuitive based on the whole `MAPPED_ATTRIBUTES` thing.

**Review by Zhao-Andy — APPROVED** [review]
LGTM!

**Pull request merged**